.travis.yml

```diff
@@ -31,10 +42,7 @@ before_install:
     - make
     - make appstore
     # install core
-    - cd ../
-    - git clone https://github.com/owncloud/core.git --recursive --depth 1 -b $CORE_BRANCH owncloud
-    - if [[ "$CORE_BRANCH" == 'master' ]]; then cd owncloud; make; cd ../; fi
-    - mv oauth2 owncloud/apps/
+    - bash tests/.travis/before_install.sh $APP_NAME $CORE_BRANCH $DB
```

Comment:
Maybe this is useful to you too: Instead of having to set `$APP_NAME` explicitly you could parse `$TRAVIS_BUILD_DIR`, which is set automatically. I guess that this could increase generality of your before_install script. :)